### ISSUE TITLE
[TASK] Callback linker to register and unregister callbacks

### ISSUE THREAD

**Issue by titasg** [Ace-Plays-Blackjack/ace-plays-game#54]
### Description

In order to adhere with S.O.L.I.D guidelines.
Create a callback linker class that gets inherited by other classes.

**Comment by titasg** [comment]
Made a universal Callback Linker that passes AcePlaysUtils type struct:

```
struct AcePlaysUtils{
    cv::Mat nextFrame;
    Card_params cardParams;
    decisions blackjackDecision;
};
```

**Comment by titasg** [comment]
Addressed in #73
